### PR TITLE
some s2 pps

### PR DESCRIPTION
**Pull request by lekoOwO**

### CODE REVIEW

File: server/pre-processors/name-english-filter.s2.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+function filter(data, errorcb) {
+    if (!data) return false;
+    let [name, id] = data;
+
+    for (const char of name) {
```
Comment (seadog007) [inline server/pre-processors/name-english-filter.s2.js]: why you doing this

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/pre-processors/id-validate.s2.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      sum += v * Mx[i];
+    }
+    return sum % 10 == 0;
+}
+
+function filter(data, errorcb) {
+    if (!data) return false;
+    let [name, id] = data;
+
+    if (!isValid(id)) {
```
Comment (seadog007) [inline server/pre-processors/id-validate.s2.js]: why not

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/pre-processors/name-english-filter.s2.js (hunk 1 above)
Comment (lekoOwO) [reply server/pre-processors/name-english-filter.s2.js]: k fixed

File: server/pre-processors/name-english-filter.s2.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+function filter(data, errorcb) {
+    if (!data) return false;
+    let [name, id] = data;
+    if (name.match(/^((?![A-Z|a-z|ü|é]).)*$/)) {
```
Comment (seadog007) [inline server/pre-processors/name-english-filter.s2.js]: this is still bad

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/pre-processors/name-english-filter.s2.js (hunk 3 above)
Comment (lekoOwO) [reply server/pre-processors/name-english-filter.s2.js]: u can u up

Comment (seadog007) [reply server/pre-processors/name-english-filter.s2.js]: 截圖 2019-10-13 上午3 31 59
You can just match that, and also you don't need | in regex

File: server/pre-processors/id-validate.s2.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+let tab = "ABCDEFGHJKLMNPQRSTUVXYWZIO"                     
+let A1 = new Array (1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,3,3,3,3,3,3 );
+let A2 = new Array (0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5 );
+let Mx = new Array (9,8,7,6,5,4,3,2,1,1);
+
+function isValid(id) {
+    if (id.length != 10 || !tab.includes(id[0])) return false;
+
+    let sum = A1[i] + A2[i] * 9;
```
Comment (seadog007) [inline server/pre-processors/id-validate.s2.js:15]: i is undefined

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/pre-processors/name-english-filter.s2.js (hunk 3 above)
Comment (lekoOwO) [reply server/pre-processors/name-english-filter.s2.js]: seems logic

**Review by seadog007 — APPROVED** [review]

**Pull request merged**